fw/t/unit/test_hpack.c

```diff
@@ -139,6 +139,83 @@ test_h2_hdr_name(TfwStr *hdr, TfwStr *out_name)
 	}
 }
 
+/** Helper to validate rbtree invariants */
+static int
+validator_walk(TfwHPackETbl *tree, TfwHPackNode *n,
+               int black_path_acc, int *black_path_len,
+			   const char *loc)
```

Comment:
lines misalignment

Reply:
@krizhanovsky I've spend whole morning trying to get rid of that boolean flag in `tfw_hpack_rbtree_find`, but it just do not work. This should be done in a separate task, because `tfw_hpack_encoder_index` calls both `tfw_hpack_rbtree_find` and `tfw_hpack_add_node` in such fashion that second call of `tfw_http_hdr_split` just cannot be avoided from there without changing huge number of other functions. Overall amount of changes goes far beyond scope of this task.
By rethinking that functions we definitely can make code significantly cleaner, and even remove redundant checks (for example, it's obvious that everything below `__tfw_hpack_encode` can never get NULL in `hdr` parameter. But unfortunately amount of changes is big and it is clearly not wise to apply them here.
IMO the root of the problem is TfwStr overusage which is almost just as "`void*`" here: you can represent dozens of invalid states with it (dup, not plain, empty) and boolean flag is just a hint about how that "`void*`" should be processed.